src/libsync/propagateuploadng.cpp

```diff
@@ -123,21 +250,39 @@ void PropagateUploadFileNG::slotPropfindIterate(const QString &name, const QMap<
     }
 }
 
+void PropagateUploadFileNG::updateRanges(quint64 start, quint64 size)
+{
+    for(auto iter = _serverRanges.begin(); iter != _serverRanges.end(); ++iter)
+    {
+        if (iter->start <= start && iter->size >= size) {
+            iter->start = start + size;
+            iter->size -= size;
```

Comment:
I don't get this logic.

Say serverRanges = [(1000, 1000)]
1) serverRanges.updateRanges(1000, 1000) -> [] ok
2) serverRanges.updateRanges(1000, 300) -> [(1300, 700)] ok
3) serverRanges.updateRanges(1300, 300) -> [(1600, 700)] bad
4) serverRanges.updateRanges(1000, 1001) -> [(1000, 1000)] bad
5) serverRanges.updateRanges(999, 1000) -> [(1000, 1000)] bad

Maybe the last three cases are guaranteed to never happen? This function definitely needs documentation, possibly error handling :/

Reply:
Last three are pretty guaranteed not to happen, since I wrote the code on the other side. Worth a note.

Reply:
I think being more explicit about what this function is supposed to do would help the reader. Like "Finds the range starting at 'start' and removes the first 'size' bytes from it. If it becomes empty, remove the range." And the code could match that (have some `ASSERT`s too). The calculation and `<=` etc currently suggests it's supposed to cover other cases too.

Reply:
I think adding a strict requirement on `iter->start == start` and the returning of the `bool` will help clear most of this up.